core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -383,59 +326,321 @@ func (l *logPollerWrapper) getCurrentCoordinators(ctx context.Context) (common.A
 		l.lggr.Errorw("LogPollerWrapper: Failed to register filters", "errorActive", errActive, "errorProposed", errProposed)
 		return
 	}
 
-	l.lggr.Debugw("LogPollerWrapper: new routes", "activeCoordinator", activeCoordinator.Hex(), "proposedCoordinator", proposedCoordinator.Hex())
+	l.lggr.Debugw("LogPollerWrapper: new routes", "activeCoordinator", activeCoordinator.address.Hex(), "proposedCoordinator", proposedCoordinator.address.Hex())
 	l.activeCoordinator = activeCoordinator
 	l.proposedCoordinator = proposedCoordinator
 
 	for _, subscriber := range l.subscribers {
-		err := subscriber.UpdateRoutes(activeCoordinator, proposedCoordinator)
+		err := subscriber.UpdateRoutes(activeCoordinator.address, proposedCoordinator.address)
 		if err != nil {
 			l.lggr.Errorw("LogPollerWrapper: Failed to update routes", "err", err)
 		}
 	}
 }
 
-func filterName(addr common.Address) string {
-	return logpoller.FilterName("FunctionsLogPollerWrapper", addr.String())
+func filterName(addr common.Address, version string) string {
+	return logpoller.FilterName("FunctionsLogPollerWrapper", addr.String(), "-v", version)
 }
 
-func (l *logPollerWrapper) registerFilters(coordinatorAddress common.Address) error {
-	if (coordinatorAddress == common.Address{}) {
+func (l *logPollerWrapper) registerFilters(coordinator coordinator) error {
+	if (coordinator.address == common.Address{}) {
 		return nil
 	}
-	return l.logPoller.RegisterFilter(
-		logpoller.Filter{
-			Name: filterName(coordinatorAddress),
-			EventSigs: []common.Hash{
-				functions_coordinator.FunctionsCoordinatorOracleRequest{}.Topic(),
-				functions_coordinator.FunctionsCoordinatorOracleResponse{}.Topic(),
-			},
-			Addresses: []common.Address{coordinatorAddress},
-		})
+
+	if strings.Contains(coordinator.typeAndVersion, "Functions Coordinator v1") {
+		return l.logPoller.RegisterFilter(
+			logpoller.Filter{
+				Name: filterName(coordinator.address, "1"),
+				EventSigs: []common.Hash{
+					functions_coordinator_1_1_0.FunctionsCoordinator110OracleRequest{}.Topic(),
+					functions_coordinator_1_1_0.FunctionsCoordinator110OracleResponse{}.Topic(),
+				},
+				Addresses: []common.Address{coordinator.address},
+			})
+	}
+
+	if strings.Contains(coordinator.typeAndVersion, "Functions Coordinator v2") {
+		return l.logPoller.RegisterFilter(
+			logpoller.Filter{
+				Name: filterName(coordinator.address, "2"),
+				EventSigs: []common.Hash{
+					functions_coordinator.FunctionsCoordinatorOracleRequest{}.Topic(),
+					functions_coordinator.FunctionsCoordinatorOracleResponse{}.Topic(),
+				},
+				Addresses: []common.Address{coordinator.address},
+			})
+	}
+
+	l.lggr.Errorw("RegisterFilters: Unsupported Coordinator version ", coordinator.typeAndVersion)
+	return errors.Errorf("RegisterFilters: Unsupported Coordinator version")
+}
+
+func oracleRequestLogTopic(coordinator coordinator) (common.Hash, error) {
+	if strings.Contains(coordinator.typeAndVersion, "Functions Coordinator v1") {
```

Comment:
could we extract this strings.Contains... statement into a helper function of some sort maybe? That might help reduce code duplication.
For example, a helper function could return the contract version 1 or 2, then we can use a switch statement to do conditional logic based on the contract version?

Reply:
Moved some things around to mitigate the SonarQube code duplication. Take another look